proto/stream_plan.proto

```diff
@@ -670,7 +673,7 @@ message StreamNode {
     HopWindowNode hop_window = 110;
     MergeNode merge = 111;
     ExchangeNode exchange = 112;
-    ChainNode chain = 113;
+    StreamScanNode stream_scan = 113;
```

Comment:
I think we have to keep the `ChainNode` for backwards compatibility.

Reply:
I did some investigation and it seems to be fine. I'll make sure of this before merging

Reply:
I also concern about the compatibility for renaming in protos. A hard choice for clean code and compatibility 😇

Reply:
> I did some investigation and it seems to be fine. From my understanding, Protobuf doesn't care about the **name** but only care about fields' type and sequence numbers (`field = <number>`) (unless encoded into JSON, which is not our case).
> 
> Here, only name is changed. The GitHub action warns because it doesn't know `ChainNode` is exactly `StreamScanNode`.
> 
> I'll make sure of this before merging.

Today I learnt. Very interesting, so renaming changes will generally be backwards compatible. SGTM 👍

Reply:
The test passed https://buildkite.com/risingwavelabs/pull-request/builds/34839

Reply:
> so renaming changes will generally be backwards compatible

As long as we don't rely on the field name in other places, like ser/de it with JSON or use the field name as etcd's key (like system parameters).